pkg/test/context.go

```diff
@@ -85,3 +100,38 @@ func (ctx *TestCtx) CleanupNoT() {
 func (ctx *TestCtx) AddFinalizerFn(fn finalizerFn) {
 	ctx.CleanUpFns = append(ctx.CleanUpFns, fn)
 }
+
+// CreateWithFinalizer uses the dynamic client to create an object and then adds a
+// finalizer function to delete it when Cleanup is called. In addition to the standard
+// controller-runtime client options
+func (ctx *TestCtx) CreateWithFinalizer(gCtx goctx.Context, obj runtime.Object, cleanupOptions *CleanupOptions) error {
```

Comment:
After discussing it offline we're going to rename Finalizer to Cleanup in the context package. We're using a mix of both Finalizer and Cleanup already so we need to make the API consistent.
Plus Finalizers are already a k8s concept which might cause confusion on what the context cleanup is doing here.

So change this to `CreateWithCleanup`